src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -3588,12 +4135,15 @@ void QgsOgrProvider::open( OpenMode mode )
   }
 
   // For debug/testing purposes
+  QString s_open_mode = "read-only";
   if ( !mValid )
-    setProperty( "_debug_open_mode", "invalid" );
+    s_open_mode = "invalid";
   else if ( mWriteAccess )
-    setProperty( "_debug_open_mode", "read-write" );
-  else
-    setProperty( "_debug_open_mode", "read-only" );
+    s_open_mode = "read-write";
+  setProperty( "_debug_open_mode", s_open_mode );
+#if 1
```

Comment:
The 'setProperty( "_debug_open_mode"' is not from me
* and is done elsewhere

I assume it is done for a purpose, thus did not change it.

The information given out at this point is very useful while testing
* so should be retained (with a '#if 0' for a release version)